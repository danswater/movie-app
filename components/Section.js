import styled from 'styled-components';

export default styled.div`
  position: relative;
  height: 100%;
  margin-top: ${props => (props.marginTop ? '2rem' : '0')};
  margin-bottom: ${props => (props.marginBottom ? '2rem' : '0')};
  p {
    font-size: 20px;
  }
`;
