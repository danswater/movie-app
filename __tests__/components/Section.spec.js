/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Section from '../../components/Section';

describe('Components', () => {
  describe('Image', () => {
    it('should render section with default styles', () => {
      const tree = renderer.create(
        <Section />,
      ).toJSON();

      expect(tree).toMatchSnapshot();
      expect(tree).toHaveStyleRule('margin-top', '0');
      expect(tree).toHaveStyleRule('margin-bottom', '0');
    });

    it('should render section with margin props', () => {
      const tree = renderer.create(
        <Section
          marginTop
          marginBottom
        />,
      ).toJSON();

      expect(tree).toMatchSnapshot();
      expect(tree).toHaveStyleRule('margin-top', '2rem');
      expect(tree).toHaveStyleRule('margin-bottom', '2rem');
    });
  });
});
