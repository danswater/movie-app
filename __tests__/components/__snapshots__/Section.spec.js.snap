// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Image should render section with default styles 1`] = `
.c0 {
  position: relative;
  height: 100%;
  margin-top: 0;
  margin-bottom: 0;
}

.c0 p {
  font-size: 20px;
}

<div
  className="c0"
/>
`;

exports[`Components Image should render section with margin props 1`] = `
.c0 {
  position: relative;
  height: 100%;
  margin-top: 2rem;
  margin-bottom: 2rem;
}

.c0 p {
  font-size: 20px;
}

<div
  className="c0"
/>
`;
